server/app/routes.py

```diff
@@ -67,11 +67,16 @@ async def download_video():
 		video_id = request.args.get('videoid')
 		if len(video_id) != 11: # yt vids are 11 chars
 			return 'invalid video id', 400
-		ydl_opts = {
-			'format': 'mp4/bestaudio/best',
-			'outtmpl': './downloadedvideos/%(title)s.%(ext)s',
-		}
-		download_video = DownloadVideo(ydl_opts)
-		download_status = await download_video.download_video(video_id)
 
+		download_video = DownloadVideo()
+		download_status = await download_video.download_video(video_id)
 		return jsonify(download_status, 400) if download_status == 'DownloadError' else jsonify('', 200)
+
+	@app.route('/download-status', methods=['GET', 'POST'])
+	async def download_status():
+		video_id = request.args.get('videoid')
+		if len(video_id) != 11: 
+			return 'invalid video id', 400
+		return jsonify(DownloadVideo().download_status(video_id))
```

Comment:
should return 404 if not found

Reply:
probably doesn't matter in this case actually, since front end will be checking `status` property of response anyway.